### PR TITLE
Improve logic in surface layer extraction method in GriddedData

### PR DESCRIPTION
**Pull request by jgliss**

### CODE REVIEW

File: pyaerocom/griddeddata.py [hunk 1]
```diff
@@ -1305,7 +1305,14 @@ def _infer_index_surface_level(self):
         except DimensionOrderError:
             self.reorder_dimensions_tseries()
         cname = self.dimcoord_names[-1]
+        coord = self[cname]
         from pyaerocom import vert_coords as vc
+        if 'positive' in coord.attributes:
+            if coord.attributes['positive'] == 'up':
```
Comment (hansbrenna) [inline pyaerocom/griddeddata.py:1311]: I'm not sure I understand the logic here.

**Review by hansbrenna — COMMENTED** [review]
I didn't completely understand the new logic you added

File: pyaerocom/griddeddata.py (hunk 1 above)
Comment (hansbrenna) [reply pyaerocom/griddeddata.py:1311]: i think I got it now.

Comment (jgliss) [reply pyaerocom/griddeddata.py:1311]: This relies on the assumption that the entries of the 1D lev coordinate are representative, i.e. if it is pressure levels, then `positive` should be `down`, if it is altitudes above sea level then `positive` should be `up`. However, the order of the indices in the model output can be both, sometimes the first index corresponds to surface, but for pressure coordinates, often the first index in the array is ToA.

Comment (jgliss) [reply pyaerocom/griddeddata.py:1311]: But these things always confuse me as well...

Comment (jgliss) [reply pyaerocom/griddeddata.py:1311]: http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#vertical-coordinate

Comment (hansbrenna) [reply pyaerocom/griddeddata.py:1311]: I see. They can be ordered both ways (in principle) so you want to first infer if the bottom level has the highest or lowest array value by using "up" or "down". Then you want to find the corresponding index for the highest/lowest value.

Comment (jgliss) [reply pyaerocom/griddeddata.py:1311]: Correct

**Pull request merged**